poker-pals/src/components/Poker.vue

```diff
@@ -112,125 +103,107 @@ export default {
-                this.players[i].bet = seatStates[i].bet;
-                this.players[i].accountName = seatStates[i].name;
-                this.players[i].chipTotal = seatStates[i].chips;
-                this.players[i].action = seatStates[i].action;
-                this.players[i].timer = seatStates[i].turn;
-                this.players[i].accountImage.src = this.imageFiles.find(file => file.name === seatStates[i].icon).src;
-            }
-            for(let i = 0; i < receivedCommunityCards.length; i++){
-                this.communityCards[i].src = this.imageFiles.find(file => file.name === receivedCommunityCards[i]).src;
-            }
-            this.potTotal = msg.potTotal;
-
-            // on each tablestate update, this code will run to see if: the player is seated, it is the players turn, the check/fold is toggled
-            // if all these conditions are meet then the code will send a check/fold decision without the player's input
-            // the server will either check or fold for the player automatically
-            if(this.seatID !== '' && this.players[this.seatID].timer === true && this.checkFold){
-                // the player wants to check, if can't then fold
-                this.makeDecision('CHECK/FOLD');
-            }
-        });
-
-        this.socket.on('joinRoom', msg => { // each player upon joining a room will receive the id of the seat they are to sit at
-            this.seatID = msg.seatID;
-            this.tableName = msg.tableName;
-            this.roomID = msg.roomID;
-            this.bigBlind = msg.bigBlind;
-            this.players[this.seatID].youTag = true;
-        });
-
-        this.socket.on('leaveRoom', () => { // each player upon joining a table will receive the id of the seat they are to sit at
-            this.seatID = -1;
-            this.tableName = '';
-            this.roomID = -1;
-            this.bigBlind = 0;
-            for(let i = 0; i < 6; i++){
-                let player = this.players[i];
-                player.occupied = false;
-                player.dealerStatus = false;
-                player.cards = {card1: {src: null}, card2: {src: null}};
-                player.bet = 0;
-                player.accountName = "";
-                player.accountImage.src = this.imageFiles.find(file => file.name === 'invisible').src;
-                player.chipTotal = "";
-                player.action = "WAITING";
-                player.youTag = false;
-                player.timer = false;
-            }
-            this.$router.push({ name: "Tables" });
-        });
+        if(!this.authenticated){
+            this.$router.replace({ name: "Login" });
+        }
+        else{
+            // define all of the socket.on methods here
+            this.socket.on('tableState', msgJSON => { // on a state update the UI is updated with all relavent info that can change, this simplifies socket calls by not having a unique socket call for each type of element update
```

Comment:
Why are these socket methods inside `else`? If the user isn't authenticated, shouldn't we leave this page?

Reply:
we are leaving this page, but for a short moment on routing to this page from tables.vue it still receives the socket messages before checking the authentication

Reply:
not sure if its a bug but this was a quick fix

Reply:
Oh, okay. Thanks for the explanation!